### PR TITLE
rosie id: accept ~/cylc-run/SUITE as argument

### LINKED ISSUE #1534: rosie id: extend to suite run directory?
Could/should we extend `rosie id` to give meaningful results when invoked within a
running suite context? For example:

```
rosie id ~/cylc-run/mi-aa123
```

would give:

```
mi-aa123/trunk@4567
```

(We'd also need to extend it to have an option to actually output the branch and revision
as well as the id).

### PR DESCRIPTION
**Pull request by matthewrmshin**
(Also works if `PWD` is `~/cylc-run/SUITE` or a sub-directory of.)

Close #1534.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@benfitzpatrick please review.

File: t/rosie-id/00-basic.t [hunk 1]
```diff
@@ -138,6 +138,27 @@ file_cmp "$TEST_KEY.out" "$TEST_KEY.out" <<__OUT__
...
+    [[t1]]
+__SUITE_RC__
+rose suite-run -l -q -C "${PWD}/foo-aa000" --name="${SUITE_NAME}"
+run_pass "$TEST_KEY" rosie id "${HOME}/cylc-run/${SUITE_NAME}"
+file_cmp "$TEST_KEY.out" "$TEST_KEY.out" <<__OUT__
+foo-aa000
+__OUT__
+file_cmp "$TEST_KEY.err" "$TEST_KEY.err" </dev/null
+rose suite-clean -y --name="${SUITE_NAME}"
+rm -fr 'foo-aa000'
```
Comment (benfitzpatrick) [inline t/rosie-id/00-basic.t]: `fr`? `rf` above

**Comment by matthewrmshin** [comment]
I have tidied up the `rm -fr` syntax.

File: lib/python/rosie/suite_id.py [hunk 2]
```diff
@@ -352,9 +363,39 @@ def _from_id_text(self, id_text):
...
+        # Is location a "~/cylc-run/$SUITE/" directory?
+        # Use a hacky way to read the "log/rose-suite-run.version" file
+        suite_engine_proc = SuiteEngineProcessor.get_processor()
+        suite_dir_rel_root = getattr(
+            suite_engine_proc, "SUITE_DIR_REL_ROOT", None)
+        if suite_dir_rel_root and "/" + suite_dir_rel_root + "/" in location:
+            loc = location
+            while "/" + suite_dir_rel_root + "/" in loc:
+                suite_version_file_name = os.path.join(
+                    loc, "log/rose-suite-run.version")
```
Comment (benfitzpatrick) [inline lib/python/rosie/suite_id.py]: A bit cylc-specific, but it is really not likely to matter.

**Pull request merged**